mlflow/langchain/langchain_tracer.py

```diff
@@ -238,6 +258,10 @@ def on_llm_start(
         """Run when LLM (non-chat models) starts running."""
         if metadata:
             kwargs.update({"metadata": metadata})
+
+        mlflow_messages = [ChatMessage(role="user", content=prompt) for prompt in prompts]
+        kwargs.update({SpanAttributeKey.CHAT_MESSAGES: mlflow_messages})
```

Comment:
looking at the examples in test cases, i feel it's a bit confusing for users for `tool_calls` and `tool_call_id` to always be present, when most of the time they will be `None`. can we use `exclude_unset=True` when serializing?

i think the only potential grey area here is `content`, because that field should always be present (but is None in the case of refusal or tool call). however, it looks like when we instantiate `ChatMessage` we always set the content, even if it is `None`, so `exclude_unset` should cover this case.

Reply:
filed https://github.com/mlflow/mlflow/pull/14155 to fix this for openai

Reply:
Yea using `exclude_unset` sounds good to me! OAI does the same for their response; they don't return `tool_calls: None` for non tool calling request.